plugins/plugin-pullrequest-parent/che-plugin-pullrequest-ide/src/main/java/org/eclipse/che/plugin/pullrequest/client/steps/InitializeWorkflowContextStep.java

```diff
@@ -100,7 +100,24 @@ public void apply(final List<Remote> remotes) throws OperationException {
         };
     }
 
-    private String getContributeToBranchName(final ProjectConfig project) {
+    protected void setContributeToBranchName(Context context) {
+        String contributeToBranchName = getBranchFromProjectMetadata(context.getProject());
+
+        if (contributeToBranchName != null) {
+            context.setContributeToBranchName(contributeToBranchName);
+            return;
+        }
+
+        vcsServiceProvider.getVcsService(context.getProject()) //
```

Comment:
Can you elaborate what's wrong with that ?
If the default formatting rules in Che doesn't let the Che users format their code the way they want then we are going nowhere: https://github.com/eclipse/che/issues/339